lib/dispatcher.js

```diff
@@ -151,7 +151,7 @@ Dispatcher.prototype.dispatch = function(params, dispatchOptions) {
         me.authenticator.authenticateRequest(params);
```

Comment:
Hmm, interesting. It took me a second to grok how the headers are treated on these lines, but I'm OK with the behavior. What happens here is that we set the auth header...